unit_test/hip/Test_HIP_Batched_SerialEigendecomposition_Real.cpp

```diff
@@ -0,0 +1,3 @@
+#include "Test_HIP.hpp"
```

Comment:
All these test files needed ? I am worried we are going in the direction of the ETI again. This is going to blow up with OpenMP target and Intel SYCL. Can we do same trick as we did for ETI, where we generate these files if unit tests are ON ?

Reply:
Yes something like that or having all the headers for the different backends in a single file and use guards to enable them selectively at compile time with different executably names.

Reply:
see issue #845